Ops/Expr/XML.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Xml.Linq;
+
+namespace PSI;
+
+public class ExprXML : Visitor<XElement> {
+
+   public ExprXML (string expression)
+      => mExpression = expression;
+
+   public override XElement Visit (NLiteral literal) {
+      var node = new XElement ("Literal");
+      node.SetAttributeValue ("Value", literal.Value.Text);
+      node.SetAttributeValue ("Type", literal.Type);
+      mLines.AppendLine (node.ToString ());
+      return node;
+   }
+
+   public override XElement Visit (NIdentifier identifier) {
+      var node = new XElement ("Ident");
+      node.SetAttributeValue("Name",identifier.Name);
+      node.SetAttributeValue ("Type", identifier.Type);
+      mLines.AppendLine (node.ToString ());
+      return node;
+   }
+
+   public override XElement Visit (NUnary unary) {
+      var node = new XElement ("Unary");
+      node.SetAttributeValue ("Op", unary.Op.Kind);
+      node.SetAttributeValue ("Type", unary.Type);
+      var node1 = unary.Expr.Accept (this);
+      node.Add (node1);
+      mLines.Clear (); 
+      mLines.Append (node.ToString ());
+      return node;

```

Comment:
The XElement constructor can take attributes, child nodes etc all as parameters. So you should be able to construct the entire XElement node for this in a single statement using code like this:
`node = new XElement ("Unary", new XAttribute ("Op", unary.Op.Kind), unary.Expr.Accept (this));`